### PR TITLE
Implement replying to annotations

### LINKED ISSUE #870: Properly track parent posts when commenting and annotating submissions

### PR DESCRIPTION
**Pull request by lowjoel**
Fixes #870.

### CODE REVIEW

**Comment by lowjoel** [comment]
I can't figure out why Travis keeps failing this. I cannot reproduce the failure on my machine.

**Comment by allenwq** [comment]
Now it passed, it's random or something ?

**Comment by lowjoel** [comment]
Like this also: https://travis-ci.org/Coursemology/coursemology2/builds/112814748

**Comment by lowjoel** [comment]
@allenwq might be a PhantomJS 1.x bug, said to be fixed in 2.0?

File: spec/features/course/assessment/submission/programming_answer_comment_spec.rb [hunk 1]
```diff
@@ -15,6 +15,18 @@
 
     context 'As a course student' do
       let(:user) { student }
+      let(:answers) { assessment.questions.attempt(submission) }
+      def annotation
```
Comment (allenwq) [inline spec/features/course/assessment/submission/programming_answer_comment_spec.rb]: @lowjoel Why not just use a let block ?  I don't think defining method in specs is good..

File: spec/features/course/assessment/submission/programming_answer_comment_spec.rb [hunk 2]
```diff
@@ -69,16 +81,29 @@ def test:
...
-      scenario 'I can delete my annotations', js: true do
-        answers = assessment.questions.attempt(submission)
-        file = answers.first.actable.files.first
-        file.content = "test\n\n\n\nlines"
-        answers.each(&:save!)
-        submission.finalise!
-        submission.publish!
-        submission.save!
+      scenario 'I can reply to an existing annotation', js: true do
+        annotation = self.annotation
```
Comment (allenwq) [inline spec/features/course/assessment/submission/programming_answer_comment_spec.rb]: @lowjoel Do we still need this ?

File: spec/features/course/assessment/submission/programming_answer_comment_spec.rb [hunk 3]
```diff
@@ -69,16 +81,29 @@ def test:
...
+        end
 
-        annotation = file.annotations.build(line: 1)
+        wait_for_ajax
+        new_post = post.children.reload.last
+        expect(new_post.text).to eq(annotation_text)
+      end
+
+      scenario 'I can delete my annotations', js: true do
+        annotation = self.annotation
```
Comment (allenwq) [inline spec/features/course/assessment/submission/programming_answer_comment_spec.rb]: and here also

**Comment by lowjoel** [comment]
@allenwq fixed, thanks.

**Pull request merged**